cypress/e2e/article.cy.js

```diff
@@ -1,24 +1,184 @@
+  });
+
+  it('The user can create new article using only title', () => {
+    const { articleTitle, articleDescription, articleBody, articleTags }
+      = generateNewArticle();
+
+    cy.findByPlaceholder('Article Title').type(articleTitle);
+    cy.contains('button', 'Publish Article').click();
+    cy.wait(2000);
+    cy.get('h1').should('contain.text', articleTitle);
+  });
+
+  it('The user can create new article using title and description', () => {
+    const { articleTitle, articleDescription, articleBody, articleTags }
+      = generateNewArticle();
+
+    cy.findByPlaceholder('Article Title').type(articleTitle);
+    cy.findByPlaceholder(`What's this article about?`).type(articleDescription);
+    cy.contains('button', 'Publish Article').click();
+  });
+
+  it('The user can create new article using all field without Tags ', () => {
+    const { articleTitle, articleDescription, articleBody, articleTags }
+      = generateNewArticle();
+
+    cy.findByPlaceholder('Article Title').type(articleTitle);
+    cy.findByPlaceholder(`What's this article about?`).type(articleDescription);
+    cy.findByPlaceholder(`Write your article (in markdown)`).type(articleBody);
+    cy.contains('button', 'Publish Article').click();
+  });
+
+  it(`The user can't create new article without title`, () => {
+    const { articleTitle, articleDescription, articleBody, articleTags }
+      = generateNewArticle();
+
+    cy.findByPlaceholder(`What's this article about?`).type(articleDescription);
+    cy.findByPlaceholder(`Write your article (in markdown)`).type(articleBody);
+    cy.contains('button', 'Publish Article').click();
+    cy.contains('.swal-modal', 'Oops!').should('exist');
   });
+});
 
+describe(`Edit article`, () => {
   beforeEach(() => {
     cy.task('db:clear');
+    cy.registerNewUser().then(user => {
+      cy.login(user).then(() => {
+        cy.createNewArticle(user).then((slug) => {
+          cy.visit(`#/articles/${slug}`);
+          user.articleSlug = slug;
+        });
+      });
+    }).as('user');
   });
 
-  it('should be created using New Article form', () => {
+  it('The user can edit whole of their own article.', () => {
+    cy.get('@user').then(user => {
+      const newTitle = faker.lorem.sentence(2).slice(0, -1);
+      const newDescription = faker.lorem.sentence(2).slice(0, -1);
+      const newBody = faker.lorem.sentence(2).slice(0, -1);
+      const newTag = faker.lorem.sentence(1).slice(0, -1);
 
+      cy.contains('a', 'Edit Article').should('exist').click();
+      cy.findByPlaceholder('Article Title').clear().type(newTitle);
+      cy.findByPlaceholder(`What's this article about?`).clear().type(newDescription);
+      cy.findByPlaceholder(`Write your article (in markdown)`).clear().type(newBody);
+      cy.findByPlaceholder(`Enter tags`).type(`${newTag}{enter}`);
+      cy.contains('button', 'Publish Article').should('exist').click();
+    });
   });
 
-  it('should be edited using Edit button', () => {
```

Comment:
Use test design techniques for article editing. No need to duplicate tests. (You checked that the title could be edited in the previous tests. Check also inputs using EV and BV